tests/expose_test.go

```diff
@@ -536,47 +536,99 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				return vmi.Status.Phase == v1.Running
 			}, 120*time.Second, 1*time.Second).Should(BeTrue())
 
-			tests.GenerateHelloWorldServer(vmi, testPort, "tcp")
-		})
+			return vmi
+		}
+
+		createStoppedVM := func(virtClient kubecli.KubevirtClient, namespace string) error {
+			By("Creating an VM object")
+			template := newLabeledVMI("vm", virtClient, false)
+			vm = tests.NewRandomVirtualMachine(template, false)
+
+			By("Creating the VM")
+			vm, err = virtClient.VirtualMachine(namespace).Create(vm)
+			return err
+		}
+
+		startVMWithServer := func(protocol string, port int) *v1.VirtualMachineInstance {
+			By("Calling the start command on the stopped VM")
+			vmi := startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace())
+			if vmi == nil {
+				return nil
+			}
+			tests.GenerateHelloWorldServer(vmi, port, protocol)
+			return vmi
+		}
 
 		Context("Expose a VM as a ClusterIP service.", func() {
-			It("[test_id:1538][label:masquerade_binding_connectivity]Connect to ClusterIP service that was set when VM was offline.", func() {
+			tests.BeforeAll(func() {
+				tests.BeforeTestCleanup()
+			})
+
+			BeforeEach(func() {
+				Expect(createStoppedVM(virtClient, tests.NamespaceTestDefault)).To(Succeed(), "should create a stopped VM.")
+			})
+
+			BeforeEach(func() {
+				vmExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachine", "--namespace", vm.GetNamespace(), vm.GetName(),
+					"--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort),
+				}
+			})
+
+			AfterEach(func() {
+				Expect(jobs).NotTo(BeEmpty(), "a successful test must have stored a way to delete the batchv1.Job entity")
+				Expect(cleanupJobs(jobs)).To(BeEmpty(), "should be able to delete the multiple k8sv1.`Job` entities")
+			})
+
+			AfterEach(func() {
+				Expect(cleanupService(serviceName, vm.GetNamespace())).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			AfterEach(func() {
+				if vm != nil {
+					By("Deleting the VM ...")
+					err = virtClient.VirtualMachine(vm.GetNamespace()).Delete(vm.GetName(), &k8smetav1.DeleteOptions{})
+					Expect(err).NotTo(HaveOccurred(), "should have been able to delete the VM object")
+					Eventually(func() bool {
+						_, err := virtClient.VirtualMachine(vm.GetNamespace()).Get(vm.GetName(), &k8smetav1.GetOptions{})
+						return err != nil && errors.IsNotFound(err)
+					}, 120*time.Second, 1*time.Second).Should(BeTrue(), "the VM object should be gone")
+
+				}
+			})
+
+			table.DescribeTable("[test_id:1538][label:masquerade_binding_connectivity]Connect to ClusterIP service that was set when VM was offline.", func(ipFamily k8sv1.IPFamily) {
+				By("Exposing a service to the VM using virtctl")
+				virtctl := tests.NewRepeatableVirtctlCommand(getVirtctlExposeCommand(vmExposeArgs, ipFamily)...)
+				Expect(virtctl()).ToNot(HaveOccurred(), "should succeed exposing a service via `virtctl expose ...`")
+
+				vmi := startVMWithServer("tcp", testPort)
```

Comment:
Because in the VM tests, in one of the tests I have to create a new VM (since the last test effectively deletes the VM object).

Since I have to re-create it once, might as well do it more times (I've tried to play around with the contexts - but I didn't like how it read in the end).